### ISSUE TITLE
Allow \Neutron\TemporaryFilesystem\Manager::clean() to be configured

### ISSUE THREAD

**Issue by kenianbei** [PHP-FFMpeg/PHP-FFMpeg#573]
| Q              | A
| -------------- | ---
| Bug?           | maybe
| New Feature?   | no
| Version Used   | v.0.12
| FFmpeg Version | 4.0
| OS             | Linux

#### Actual Behavior

In \FFMpeg\Media\Video::save() method, line 106:

`$this->fs->clean($this->fsId);`

Doesn't allow passing in of $throwException option to Neutron\TemporaryFilesystem\Manager::clean() during file cleanup. This is causing a docker based workflow of mine to fail at unpredictable times.

#### Expected Behavior

Allow suppression of file cleanup errors by adding a passed in variable.

#### Steps to Reproduce

Hard to reproduce in actual use, but just add to Neutron\TemporaryFilesystem\Manager::cleanScope() for testing:

`throw new SfIOException('');`

#### Possible Solutions

It would be very easy to just add to \FFMpeg\Media\Video::save() like so:

```php
    /**
     * Exports the video in the desired format, applies registered filters.
     *
     * @param FormatInterface   $format
     * @param string            $outputPathfile
     * @param bool              $throwException
     * @return Video
     * @throws RuntimeException
     */
    public function save(FormatInterface $format, $outputPathfile, $throwException = true)
    {
        $passes = $this->buildCommand($format, $outputPathfile);

        $failure = null;
        $totalPasses = $format->getPasses();

        foreach ($passes as $pass => $passCommands) {
            try {
                /** add listeners here */
                $listeners = null;

                if ($format instanceof ProgressableInterface) {
                    $filters = clone $this->filters;
                    $duration = 0;

                    // check the filters of the video, and if the video has the ClipFilter then
                    // take the new video duration and send to the
                    // FFMpeg\Format\ProgressListener\AbstractProgressListener class
                    foreach ($filters as $filter) {
                        if($filter instanceof ClipFilter){
                            $duration = $filter->getDuration()->toSeconds();
                            break;
                        }
                    }
                    $listeners = $format->createProgressListener($this, $this->ffprobe, $pass + 1, $totalPasses, $duration);
                }

                $this->driver->command($passCommands, false, $listeners);
            } catch (ExecutionFailureException $e) {
                $failure = $e;
                break;
            }
        }

        $this->fs->clean($this->fsId, $throwException);

        if (null !== $failure) {
            throw new RuntimeException('Encoding failed', $failure->getCode(), $failure);
        }

        return $this;
    }
```

**Comment by jens1o** [comment]
I think this should throw an exception at any given time. No user manipulation required.

> This is causing a docker based workflow of mine to fail at unpredictable times.

Could you further elaborate on this? Maybe the docker image is broken?

**Comment by kenianbei** [comment]
Thanks for the quick reply... and sorry for the somewhat vague error report.

I hadn't looked closely enough at how the Video save class worked, and was somewhat confused on the error handling. I looked at it this morning and see now that it's perfectly fine.

I solved my issue just by wrapping the save function as so:

```php
        try {
            $video->save($format, $output);
        }
        catch (IOException $e) {
            error_log($e->getMessage());
        }
```
Since I don't care about the clean up, and since my docker file system is temporary anyway, I can ignore clean up errors.

With my docker workflow, for some reason on AWS I think the file system perms get changed or temp files get automatically deleted and causes the clean function to throw an error when it tries to clean up files that already don't exist. I'm still not exactly sure why it's happening.

Anyway, issue closed and thanks again for this great module, I've been using it for years.
